src/Sylius/Bundle/CoreBundle/Applicator/CatalogPromotionApplicator.php

```diff
@@ -22,13 +23,18 @@ final class CatalogPromotionApplicator implements CatalogPromotionApplicatorInte
 {
     public function __construct(
         private ActionBasedDiscountApplicatorInterface $actionBasedDiscountApplicator,
-        private ProductVariantForCatalogPromotionEligibilityInterface $checker
+        private ProductVariantForCatalogPromotionEligibilityInterface $checker,
+        private CatalogPromotionEligibilityCheckerInterface $catalogPromotionEligibilityChecker
     ) {}
 
     public function applyOnVariant(
         ProductVariantInterface $variant,
         CatalogPromotionInterface $catalogPromotion
     ): void {
+        if (!$this->catalogPromotionEligibilityChecker->isCatalogPromotionEligible($catalogPromotion)) {
```

Comment:
We are not executing `isCatalogPromotionEligibleOperatingTime`, while I'm quite sure we should. Do we have a test for this case?